packages/pn-pa-webapp/src/navigation/VerifyUser.tsx

```diff
@@ -5,40 +5,40 @@ import { exchangeToken } from '../redux/auth/actions';
 import { useAppDispatch, useAppSelector } from '../redux/hooks';
 import { RootState } from '../redux/store';
 import { getHomePage } from '../utils/role.utility';
-import { goToSelfcareLogin } from './navigation.utility';
 
 const VerifyUser = () => {
   const location = useLocation();
   const [selfCareToken, setSelfCareToken] = useState('');
   const dispatch = useAppDispatch();
   const token = useAppSelector((state: RootState) => state.userState.user.sessionToken);
   const navigate = useNavigate();
+  const [verificationDone, setVerificationDone] = useState(false);
 
   useEffect(() => {
     const params = new URLSearchParams(location.hash);
     const selfCareTokenParam = params.get('#selfCareToken');
     if (selfCareTokenParam) {
       setSelfCareToken(selfCareTokenParam);
     } else {
-      if (token === '') {
-        goToSelfcareLogin();
-      }
+      setVerificationDone(true);
     }
   }, [location]);
 
   useEffect(() => {
     if (selfCareToken !== '') {
-      dispatch(exchangeToken(selfCareToken))
-        .then(() => {
-          navigate(getHomePage(), {replace: true});
-        })
-        .catch(() => {
-          goToSelfcareLogin();
-        });
+      void dispatch(exchangeToken(selfCareToken)).then(() => {
+        setVerificationDone(true);
+      });
     }
   }, [selfCareToken]);
 
-  return <Outlet />;
+  useEffect(() => {
+    if (token !== '') {
+      navigate(getHomePage(), {replace: true});
+    }
+  }, [token]);
+
+  return verificationDone ? <Outlet /> : <div />;
```

Comment:
Se verificationDone è false ( questo può accadere anche se l'exchangeToken va in errore ) cosa succede?

Reply:
Se `exchangeToken` va in errore nel senso che lo `status` della response è 4xx oppure 5xx, il dispatch non va in errore, ma invece viene eseguito il reducer `.rejected`. Perciò, per qualsiasi status nella response, quando questa response arriva, `verificationDone` si setta a true.  
L'unico caso veramente non previsto (e che confesso che mi sono dimenticato) è quello in cui il dispatch va in errore per ragioni (di errore di sorgente) intrinseche al FE. In quel caso ... non saprei cosa fare, perché non so come vengono gestite le exception che sono dovute a errori di programmazione.

Reply:
@clombardi nell'ultimo caso da te descritto abbiamo per ora un ErrorBoundary che cattura le eccezioni js